engine/storage/src/org/apache/cloudstack/storage/datastore/ObjectInDataStoreManagerImpl.java

```diff
@@ -92,6 +92,7 @@ public ObjectInDataStoreManagerImpl() {
         stateMachines.addTransition(State.Allocated, Event.OperationFailed, State.Failed);
         stateMachines.addTransition(State.Creating, Event.OperationFailed, State.Allocated);
         stateMachines.addTransition(State.Creating, Event.OperationSuccessed, State.Ready);
+        stateMachines.addTransition(State.Creating, Event.DestroyRequested, State.Destroying);
```

Comment:
This is actually a bug that was hit while running Marvin. A Vm that is being created threw exception when destroyed as there is not respective state transition.